search_agg_percentiles.go

```diff
@@ -71,88 +71,49 @@ type PercentilesAggregationResult struct {
 
 // UnmarshalJSON implements [json.Unmarshaler] to decode a json byte slice into a PercentilesAggregationResult
 func (p *PercentilesAggregationResult) UnmarshalJSON(b []byte) error {
-	// map[key] -> value
-	var rawResp map[string]json.RawMessage
-	if err := json.Unmarshal(b, &rawResp); err != nil {
-		return err
+	type valuesJSON struct {
+		Values struct {
+			P1        *float64 `json:"1.0,omitempty"`
+			P1String  string   `json:"1.0_as_string,omitempty"`
+			P5        *float64 `json:"5.0,omitempty"`
+			P5String  string   `json:"5.0_as_string,omitempty"`
+			P25       *float64 `json:"25.0,omitempty"`
+			P25String string   `json:"25.0_as_string,omitempty"`
+			P50       *float64 `json:"50.0,omitempty"`
+			P50String string   `json:"50.0_as_string,omitempty"`
+			P75       *float64 `json:"75.0,omitempty"`
+			P75String string   `json:"75.0_as_string,omitempty"`
+			P95       *float64 `json:"95.0,omitempty"`
+			P95String string   `json:"95.0_as_string,omitempty"`
+			P99       *float64 `json:"99.0,omitempty"`
+			P99String string   `json:"99.0_as_string,omitempty"`
+		}
```

Comment:
I would have expected to need  an explicit JSON tag here but I guess `encoding/json` matches up "values" to `Values` when there's no tag.